src/Gaffer/Animation.cpp

```diff
@@ -272,7 +272,7 @@ Animation::KeyPtr Animation::Key::setTime( float time )
 
 void Animation::Key::setValue( float value )
 {
-	if( value == m_value )
+	if( Animation::equivalentValues( value, m_value ) )
```

Comment:
This is very unusual for a setter method in the Gaffer API - I can't think of an quivalent example in the rest of the API. I think I mentioned this before, but the general expectation is `setFoo( x ); assert( getFoo() == x )`. This suggests to me that if we do need this soft comparison in some places, we should do it on the client side.